File: skiecommenrce-backed/skiecommenrce-backed/src/main/java/pl/opalka/skiecommenrcebacked/dto/Purchase.java
package pl.opalka.skiecommenrcebacked.dto;

import lombok.Getter;
import lombok.Setter;
import pl.opalka.skiecommenrcebacked.model.Address;
import pl.opalka.skiecommenrcebacked.model.Customer;
import pl.opalka.skiecommenrcebacked.model.OrderDetail;
import pl.opalka.skiecommenrcebacked.model.OrderItem;

import java.util.Set;

@Getter
@Setter
public class Purchase {
    private Customer customer;
    private Address address;
    private Set<OrderItem> orderItem;
    private OrderDetail orderInfo;
    private String paymentMethod;
    private String shippingMethod;
}

File: angular-skiecommerce/src/app/classes/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { OrderInfo } from "./order-info";
import { OrderItem } from "./order-item";
import { PaymentMethod } from "./payment-method";
import { ShippingMethod } from "./shipping-method";

export class Purchase {
    customer!: Customer;
    shippingAddress!: Address;
    orderInfo!: OrderInfo;
    orderItems!: OrderItem[];
    paymentMethod!: PaymentMethod;
    shippingMethod!: ShippingMethod;
}
